docs/proposals/vip-4.md

```diff
@@ -0,0 +1,248 @@
+---
+layout: default
+title: VIP-4 Store Lifecycle Hooks
+parent: Proposals
+permalink: /docs/proposals/vip-4
+---
+
+# VIP-4: Store Lifecycle Hooks
+
+* **Status**: _Under Discussion_
+* **Author(s)**: _Felix GV_
+* **Pull Request**: _N/A_
+* **Release**: _N/A_
+
+## Introduction
+
+The [Venice Push Job](../user_guide/write_api/push_job.md) takes data from a grid and pushes it to a store in all 
+regions. This works fine in many cases, but there are some use cases where we would like to have greater control over 
+the steps of the process. This proposal is to add new configs and hooks which can be used both to monitor and control 
+the push job in a finer-grained manner than is possible today. In particular, this proposal focuses on the way that each 
+individual region is handled.
+
+Currently, the sequence of steps happening within a push job is as follows:
+
+![Push Job Steps](../assets/images/push_job_steps.drawio.svg)
+
+A few notes on the above diagram:
+
+* A new schema will be registered only if the data being pushed does not conform to any already known schema, and 
+  the schema auto-registration config is enabled. If that config is disabled, then an unknown schema leads to job 
+  failure.
+
+* The compression job is optional, and whether it runs or not depends on configurations.
+
+* The data push job writes to a store-version pub sub topic in one of the regions (typically the one which is local to
+  where the push job is running), but all regions start ingesting right away, as soon as the data push job begins 
+  writing. For regions that are remote from the topic the data push job is writing to, the leader servers are performing 
+  replication, while in the region which contains that topic, the replication is a no-op.
+
+* The `SERVING` step is also called "version swap". If there are Da Vinci Clients, they will ingest and swap on their
+  own, and the child controller of that region will wait until all DVC instances have swapped (started serving) before
```

Comment:
Is this true?
I recalled DaVinci will report completed, the switch will wait for the version switch decided by the Venice Controller, and it won't switch right after completion.
cc @sixpluszero

Reply:
@sixpluszero Can you confirm ^?